packages/http-client-csharp/emitter/src/lib/model.ts

```diff
@@ -99,197 +87,29 @@ export function getInputType(
-      if (usages.isUsedAs(type, UsageFlags.Output)) value = value | UsageFlags.Output;
-      usagesMap.set(name, value);
-    }
-  }
 
-  for (const op of ops) {
-    if (!op.parameters.body?.property && op.parameters.body?.type) {
-      let effectiveBodyType: Type | undefined = undefined;
-      const affectTypes: Set<string> = new Set<string>();
-      effectiveBodyType = getEffectiveSchemaType(context, op.parameters.body.type);
-      if (effectiveBodyType.kind === "Model") {
-        /* handle spread. */
-        if (effectiveBodyType.name === "") {
-          effectiveBodyType.name = `${capitalize(op.operation.name)}Request`;
-        }
-      }
-      if (effectiveBodyType.kind === "Model") {
-        /*propagate to sub models and composite models*/
-        getAllEffectedModels(effectiveBodyType, new Set<string>()).forEach((element) => {
-          affectTypes.add(element);
-        });
-      }
-      for (const name of affectTypes) {
-        appendUsage(name, UsageFlags.Input);
-      }
-    }
-    /* handle response type usage. */
-    const affectedReturnTypes: Set<string> = new Set<string>();
-    for (const res of op.responses) {
-      const resBody = res.responses[0]?.body;
-      if (resBody?.type) {
-        let returnType = "";
-        const effectiveReturnType = getEffectiveSchemaType(context, resBody.type);
-        if (effectiveReturnType.kind === "Model" && effectiveReturnType.name !== "") {
-          returnType = getTypeName(context, effectiveReturnType);
-        }
-        /*propagate to sub models and composite models*/
-        if (effectiveReturnType.kind === "Model") {
-          getAllEffectedModels(effectiveReturnType, new Set<string>()).forEach((element) => {
-            affectedReturnTypes.add(element);
-          });
-        }
-        affectedReturnTypes.add(returnType);
-        for (const name of affectedReturnTypes) {
-          appendUsage(name, UsageFlags.Output);
-        }
-      }
-      /* calculate the usage of the LRO result type. */
-      const metadata = getLroMetadata(program, op.operation);
-      if (metadata !== undefined) {
-        let bodyType: Model;
-        if (
-          op.verb !== "delete" &&
-          metadata.finalResult !== undefined &&
-          metadata.finalResult !== "void"
-        ) {
-          bodyType = metadata.finalEnvelopeResult as Model;
-          if (bodyType) {
-            getAllEffectedModels(bodyType, new Set<string>()).forEach((element) => {
-              affectedReturnTypes.add(element);
-            });
-          }
-        }
-      }
-    }
+  // TODO -- remove when https://github.com/Azure/typespec-azure/issues/1150 is resolved
+  // TCGC has a bug that sometimes the name of body type constructed back from spread parameters does not have a name, here we add a name for it.
+  if (operation && sdkType.kind === "model" && sdkType.name === "") {
+    sdkType.name = `${getOperationFullName(operation)}Request`;
```

Comment:
this should have been fixed in the latest version of typespec/TCGC.
We should remove this once we adopted the new version of typespec.